osquery/core/conversions.cpp

```diff
@@ -316,49 +315,44 @@ bool JSON::valueToBool(const rj::Value& value) {
   return false;
 }
 
-std::string base64Decode(const std::string& encoded) {
-  std::string is;
-  std::stringstream os;
-
-  is = encoded;
-  boost::replace_all(is, "\r\n", "");
-  boost::replace_all(is, "\n", "");
-  size_t size = is.size();
+std::string base64Decode(std::string encoded) {
+  boost::replace_all(encoded, "\r\n", "");
+  boost::replace_all(encoded, "\n", "");
 
   // Remove the padding characters
-  if (size && is[size - 1] == '=') {
-    --size;
-    if (size && is[size - 1] == '=') {
-      --size;
+  for (int i = 0; i < 2; i++) {
+    if (!encoded.empty() && encoded.back() == '=') {
+      encoded.pop_back();
     }
   }
 
-  if (size == 0) {
+  if (encoded.empty()) {
     return "";
   }
+
   try {
-    std::copy(base64_dec(is.data()),
-              base64_dec(is.data() + size),
-              std::ostream_iterator<char>(os));
+    return std::string(base64_dec(encoded.data()),
+                       base64_dec(encoded.data() + encoded.size()));
   } catch (const boost::archive::iterators::dataflow_exception& e) {
     LOG(INFO) << "Could not base64 decode string: " << e.what();
     return "";
   }
-  return os.str();
 }
 
 std::string base64Encode(const std::string& unencoded) {
-  std::stringstream os;
-
-  if (unencoded.size() == 0) {
+  if (unencoded.empty()) {
     return std::string();
   }
 
   size_t writePaddChars = (3U - unencoded.length() % 3U) % 3U;
-  std::string base64(it_base64(unencoded.begin()), it_base64(unencoded.end()));
-  base64.append(writePaddChars, '=');
-  os << base64;
-  return os.str();
+  try {
+    return std::string(it_base64(unencoded.begin()),
+                       it_base64(unencoded.end())) +
+           std::string(writePaddChars, '=');
```

Comment:
There are 3 memory allocations. It could be done with one :)
Smth like this:
```c++
auto enc = std::string(it_base64(unencoded.begin()), it_base64(unencoded.end()));
enc.append(writePaddChars, '=');
return enc;
```